test-framework/test-suites/integration/tests/add/test_add_pallet.py

```diff
@@ -100,6 +100,34 @@ def test_mountpoint_in_use(self, host, create_pallet_isos, revert_export_stack_p
 			}
 		]
 
+	def test_mountpoint_in_use_mnt(self, host, create_blank_iso, create_pallet_isos, revert_export_stack_pallets):
+		# Mount an ISO to simulate something left mounted
+		result = host.run(f'mount {create_blank_iso}/blank.iso /mnt')
+		assert result.rc == 0
+
+		# Add our minimal pallet
+		result = host.run(f'stack add pallet {create_pallet_isos}/minimal-1.0-sles12.x86_64.disk1.iso')
+		assert result.rc == 0
+		assert result.stdout == 'Copying minimal 1.0-sles12 to pallets ...\n'
+
+		# Check it made it in as expected
+		result = host.run('stack list pallet minimal output-format=json')
+		assert result.rc == 0
+		assert json.loads(result.stdout) == [
+			{
+				'name': 'minimal',
+				'version': '1.0',
+				'release': 'sles12',
+				'arch': 'x86_64',
+				'os': 'sles',
+				'boxes': ''
+			}
+		]
+		
+		# Unmount ISO  
+		result = host.run(f'umount /mnt')
```

Comment:
If not unmounting the iso was the source of the other test failures, it might be best to wrap this test body in a ExitStack so that cleanup operations can be guaranteed to run should an assert happen before this line is reached. 

Example:
```
from contextlib import ExitStack

def test_mountpoint_in_use_mnt(self, host, create_blank_iso, create_pallet_isos, revert_export_stack_pallets):
	with ExitStack() as cleanup:
		# Mount an ISO to simulate something left mounted
		result = host.run(f'mount {create_blank_iso}/blank.iso /mnt')
		assert result.rc == 0
		# Make sure we unmount the iso no matter how we exit this test
		cleanup.callback(host.run, 'umount /mnt')
		# Rest of test code below
		...
```

ExitStack docs: https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack

This example can't assert the return code of umount, but I'd argue that the test isn't about testing whether umount works or not.

Reply:
☝️